frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -130,61 +167,227 @@ const ShimmeringBody: React.FC<ShimmeringBodyProps> = ({
   </tr>
 );
 
-type ExpandingCellProps = {
-  index: number;
-  expandedRows: RowIndex[];
+type ExpandingButtonProps = {
+  rowKey: string;
+  expandedRows: RowKey[];
   rowValues: any;
   onClick: (index) => void;
-  onExpand?: (rowValues: any, index: number) => void;
-  onCollapse?: (rowValues: any, index: number) => void;
+  onExpand?: (rowValues: any, key: string) => void;
+  onCollapse?: (rowValues: any, key: string) => void;
   isSelectedRow: boolean;
+  nestedLevel: number;
 };
-const ExpandingCell: React.FC<ExpandingCellProps> = ({
-  index,
+const ExpandingButton: React.FC<ExpandingButtonProps> = ({
+  rowKey,
   onClick,
   onExpand,
   onCollapse,
   rowValues,
   expandedRows,
   isSelectedRow,
-}: ExpandingCellProps) => {
-  const isExpanded = expandedRows.includes(index);
-  const cellStyling = { width: EXPANDING_CELL_WIDTH };
+  nestedLevel,
+}: ExpandingButtonProps) => {
+  const isExpanded = expandedRows.includes(rowKey);
+  const buttonContainerStyle = {
+    width: `${getExpandingButtonWidth(nestedLevel)}px`,
+  };
 
   return (
-    <td
-      className="ams-table-cell ams-table-expanding-cell"
-      key={`expandingIndex:${index}`}
-      style={cellStyling}
+    <span
+      className="ams-table-expanding-button-container"
+      style={buttonContainerStyle}
     >
       <button
+        key={rowKey}
         type="button"
         className={`btn ams-table-expanding-button ${
-          isSelectedRow && 'is-selected-row'
-        }`}
+          rowValues.isNestedColumn && 'is-nested-column-row'
+        } ${isSelectedRow && 'is-selected-row'}`}
```

Comment:
This one is creating the undefined.
Let's do instead:
```suggestion
        } ${isSelectedRow ? 'is-selected-row' : ''}`}
```